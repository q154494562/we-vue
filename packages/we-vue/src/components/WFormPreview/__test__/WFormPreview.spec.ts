import { shallowMount } from '@vue/test-utils'
import FormPreview from '../'

describe('preview', () => {
  test('create', () => {
    const wrapper = shallowMount(FormPreview, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-form-preview')
    expect(wrapper.classes()).toContain('weui-form-preview')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render title and value', () => {
    const wrapper = shallowMount(FormPreview, {
      propsData: {
        title: 'test-title',
        value: 'test-value',
      },
    })

    expect(wrapper.find('label.weui-form-preview__label').text()).toBe('test-title')
    expect(wrapper.find('em.weui-form-preview__value').text()).toBe('test-value')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render dataItems', () => {
    const wrapper = shallowMount(FormPreview, {
      propsData: {
        dataItems: [
          {
            label: 'label1',
            value: 'value1',
          },
          {
            label: 'label2',
            value: 'value2',
          },
          {
            label: 'label3',
            value: 'value3',
          },
        ],
      },
    })

    expect(wrapper.findAll('.weui-form-preview__item')).toHaveLength(3)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('test buttons', () => {
    const actionSpy = jest.fn()

    const wrapper = shallowMount(FormPreview, {
      propsData: {
        buttons: [
          {
            text: 'OK',
            action: actionSpy,
          },
          {
            text: 'CANCEL',
          },
        ],
      },
    })

    expect(wrapper.findAll('.weui-form-preview__btn')).toHaveLength(2)

    expect(wrapper.html()).toMatchSnapshot()

    // click the 'OK' button
    wrapper.findAll('.weui-form-preview__btn').at(0).trigger('click')
    expect(actionSpy).toHaveBeenCalled()
  })
})
