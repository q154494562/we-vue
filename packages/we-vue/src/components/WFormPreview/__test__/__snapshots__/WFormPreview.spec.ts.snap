// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preview create 1`] = `
<div class="weui-form-preview">
  <div class="weui-form-preview__hd"><label class="weui-form-preview__label"></label><em class="weui-form-preview__value"></em></div>
  <div class="weui-form-preview__bd"></div>
  <div class="weui-form-preview__ft"></div>
</div>
`;

exports[`preview render dataItems 1`] = `
<div class="weui-form-preview">
  <div class="weui-form-preview__hd"><label class="weui-form-preview__label"></label><em class="weui-form-preview__value"></em></div>
  <div class="weui-form-preview__bd">
    <div class="weui-form-preview__item"><label class="weui-form-preview__label">label1</label><span class="weui-form-preview__value">value1</span></div>
    <div class="weui-form-preview__item"><label class="weui-form-preview__label">label2</label><span class="weui-form-preview__value">value2</span></div>
    <div class="weui-form-preview__item"><label class="weui-form-preview__label">label3</label><span class="weui-form-preview__value">value3</span></div>
  </div>
  <div class="weui-form-preview__ft"></div>
</div>
`;

exports[`preview render title and value 1`] = `
<div class="weui-form-preview">
  <div class="weui-form-preview__hd"><label class="weui-form-preview__label">test-title</label><em class="weui-form-preview__value">test-value</em></div>
  <div class="weui-form-preview__bd"></div>
  <div class="weui-form-preview__ft"></div>
</div>
`;

exports[`preview test buttons 1`] = `
<div class="weui-form-preview">
  <div class="weui-form-preview__hd"><label class="weui-form-preview__label"></label><em class="weui-form-preview__value"></em></div>
  <div class="weui-form-preview__bd"></div>
  <div class="weui-form-preview__ft">
    <div class="weui-form-preview__btn weui-form-preview__btn_default">OK</div>
    <div class="weui-form-preview__btn weui-form-preview__btn_default">CANCEL</div>
  </div>
</div>
`;
